internal/service/ec2/transitgateway_project_access.go

```diff
@@ -0,0 +1,164 @@
+package ec2
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+)
+
+const (
+	idSeparator = "/"
+)
+
+func ResourceTransitGatewayProjectAccess() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: ResourceTransitGatewayProjectAccessCreate,
+		ReadWithoutTimeout:   ResourceTransitGatewayProjectAccessRead,
+		DeleteWithoutTimeout: ResourceTransitGatewayProjectAccessDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: ResourceTransitGatewayProjectAccessImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"transit_gateway_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func ResourceTransitGatewayProjectAccessCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).EC2Conn
+
+	transitGatewayID := d.Get("transit_gateway_id").(string)
+	projectID := d.Get("project_id").(string)
+
+	modifyInput := &ec2.ModifyTransitGatewayInput{
+		TransitGatewayId: aws.String(transitGatewayID),
+		Options: &ec2.ModifyTransitGatewayOptions{
+			AddSharedOwners: []*string{aws.String(projectID)},
+		},
+	}
+
+	log.Printf("[DEBUG] Creating Transit Gateway Project Access: %s", modifyInput)
+	_, err := conn.ModifyTransitGateway(modifyInput)
+	if err != nil {
+		return diag.Errorf("error sharing Transit Gateway (%s) to project (%s): %s", transitGatewayID, projectID, err)
+	}
+
+	id := buildID(transitGatewayID, projectID)
+	d.SetId(id)
+
+	return ResourceTransitGatewayProjectAccessRead(ctx, d, meta)
+}
+
+func ResourceTransitGatewayProjectAccessRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).EC2Conn
+
+	transitGatewayID := d.Get("transit_gateway_id").(string)
+	projectID := d.Get("project_id").(string)
+
+	transitGateway, err := FindTransitGatewayByID(conn, transitGatewayID)
+
+	if !d.IsNewResource() && tfresource.NotFound(err) {
+		log.Printf("[WARN] Transit Gateway (%s) not found, removing from state", transitGatewayID)
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return diag.Errorf("error reading EC2 Transit Gateway (%s): %s", transitGatewayID, err)
```

Comment:
```suggestion
		return diag.Errorf("error reading Transit Gateway (%s): %s", transitGatewayID, err)
```

В остальных логах просто Transit Gateway